File: app/src/main/java/com/jesusc24/xroadsthroughthecastle/data/model/BugComparator.java
package com.jesusc24.xroadsthroughthecastle.data.model;


import java.util.Comparator;

public class BugComparator implements Comparator {

    //Ordena por la gravedad del bug
    @Override
    public int compare(Object o1, Object o2) {
        return ((Bug)o2).getGravedad().compareTo(((Bug)o2).getGravedad());
    }
}

File: app/src/main/java/com/jesusc24/xroadsthroughthecastle/data/model/BugComparator.kt
package com.jesusc24.xroadsthroughthecastle.data.model

class BugComparator : Comparator<Bug> {
    override fun compare(p0: Bug, p1: Bug): Int {
        val estado0 = p0.estado
        val estado1 = p1.estado
        return estado0?.compareTo(estado1.toString())!!
    }
}